### ISSUE TITLE
Scope of case studies section

### ISSUE THREAD

**Issue by olexandr-konovalov** [WSSSPE/WG-Best-Practices#4]
Which case studies do we need for the section 3? What is their subject?

Since Section 4 is on community-related practices, it seems to me that the case studies should be focused on software projects rather than on communities. Does this sound right?

If we agree on that, then how to select these projects? It would be good to establish some criteria and try to include a range of scientific software projects that varies by application domains, development models, programming environments, user base, maturity, etc.

For now, I suggest to start with collecting some suggestions for case studies as comments under this issue. These could be projects in which authors are involved or are familiar with, but not necessarily - we may establish a should questionnaire to contact developers of other project. I hope that when we will have an overview of what we could include, we will have a better idea. At the end, we will likely have to make a shortlist, being limited by page volume.

**Comment by iliant** [comment]
I can help with a portfolio of projects - http://www.ccp5.ac.uk/software/ - from mature ones, 2+ decades, to new ones, ~5 years.

Ilian

**Comment by olexandr-konovalov** [comment]
I will suggest a case study of GAP (http://www.gap-system.org/) - a system for computational discrete algebra, of which I am one of the developers. It uses the kernel written in C which provides runtime environment for the interpreted programming language, also called GAP. The project was started in 1986, and the current version is 4.8.3 with the architecture of the system rooting from the first release of GAP 4 in 1999 (prior to that, there was GAP 3). It is used mainly by mathematicians (see appplication areas in http://www.gap-system.org/Doc/Bib/statistics.html), but there are interesting applications outside too. After using CVS for yeas and then Mercurial since summer 2012, in February 2015 we migrated to Git and put the repository at https://github.com/gap-system/gap

**Comment by kyleniemeyer** [comment]
If you look at the [original outline](https://docs.google.com/document/d/1SqIsDqwhxxNvKnxCgW4ny4dZ13tkM5tFwGJOdAQpbbs/edit), there were three proposed case studies (that came out of the working group discussion at WSSSPE3): PeTSC, NWCHEM, and CIG.

**Comment by kyleniemeyer** [comment]
Although, as Dan suggested, these are fairly large projects, so perhaps they could be supplemented with a few smaller ones too.

**Comment by abani1** [comment]
The case studies should reflect the diversity of the Scientific s/w
community --
a) PeTSc -- large well supported and general use single focus general use
tool
b) NWCHEM -- large application centric community for single focus
application
c) CIG -- large focused community supporting a suite of tools with a great
model of community resourcing and governance
d) small application focused tool(s) -- nominations please
e) small general purpose tool(s) -- nominations please
f) SSE, SSI supported toolset?

d) and e) should be single group developers -- a very popular mode for
scientific tools
Thoughts?

abani

On Wed, May 25, 2016 at 2:42 AM, Kyle Niemeyer notifications@github.com
wrote:

> Although, as Dan suggested, these are fairly large projects, so perhaps
> they could be supplemented with a few smaller ones too.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/WSSSPE/WG-Best-Practices/issues/4#issuecomment-221486880

## 

Abani Patra

**Comment by JeffCarver** [comment]
abani, I agree with your suggestions here. We need a diversity of projects because we are likely to see differences among them depending on size, support, domain, whether something is a library, etc...

**Comment by dangunter** [comment]
I would distinguish between libraries (and tools) and services, in the sense of centralized vs. localized usage models. For a "library", the code repository is the only centralized component. For a "service" the deployment is also centralized, and this adds a devops community to the user and developer communities present in both.

**Comment by kyleniemeyer** [comment]
I also agree with @albani's suggestions, and would like to try and jump start some progress on this section of the paper. 

@iliant (or others, but he did volunteer on this 😄): do you have any thoughts on smaller tools?

I also think (NSF SI2) SSE/SSI supported projects could be good to look at for any additions, although we probably don't want to limit ourselves to US-led projects. Any thoughts on [SAGE](http://www.sagemath.org), numpy/scipy, or [yt](http://yt-project.org)?

**Comment by dangunter** [comment]
Perhaps pymatgen https://github.com/materialsproject/pymatgen? This
started as a single-team effort but has since accrued people from multiple
teams (partially, but not entirely, due to splintering of original team)
and volunteer contributions. It is not a large tool by any means, but it
has users, and is healthy and evolving to incorporate new methods. It is
also an important code for the Materials Project website; dual use as a
standalone tool and in a public web service is, I think, a model of
sustainability worth including (whether or not pymatgen is the exemplar).

-Dan
